### ISSUE TITLE
Feature request: Lose / modify the "chat" bubble background colors in SMS

### ISSUE THREAD

**Issue by ferdnyc** [andyholmes/gnome-shell-extension-gsconnect#53]
The "chat bubble" colors are a cute idea, but in my experience don't provide the most readable display, despite the use of white text when necessary to meet WCAG contrast standards. The problem, I think, is that the colors are taken from the standard Gnome Tango palette, and those colors were meant to be used in graphic design applications (icons and interface chrome), not for text backgrounds.

I captured this screenshot from one conversation (excuse the partial censoring) — it _technically_ passes WCAG contrast tests, but that doesn't mean it's easily readable.

![screenshot from 2018-02-01 10-40-17](https://user-images.githubusercontent.com/538020/36706446-dfc55942-1b37-11e8-9b39-70b0fa0a0a89.png)

And your own screenshot has an even worse example, because of the hyperlink coloring:

![screenshot from 2018-02-26 20-30-15](https://user-images.githubusercontent.com/538020/36706029-0abd05ca-1b36-11e8-9f90-968c4a0aa603.png)

I have a feeling the Tango color palette is just too bold to be used for this purpose. If GSConnect is going to color text bubbles, I would suggest having it used a lightened set of colors _based on_ Tango, always paired with black text. (The Gnome HIG even [offers](https://developer.gnome.org/hig/stable/icons-and-artwork.html.en), "You are free to use different shades of these colors depending on the desired material effect.") 

For example, here's the Tango palette mixed with 80% white (I can generate the corresponding `#RRGGBB` values if needed)...

![screenshot from 2018-02-26 20-47-36](https://user-images.githubusercontent.com/538020/36706430-cac510b4-1b37-11e8-85e6-244f8770d374.png)

I think something like that would be a much more comfortable palette for message background coloring.

**Comment by andyholmes** [comment]
I couldn't agree more. I initially took a much harder stance on Gnome HIG & Design compliance, but my outlook has changed quite a bit. I regret how long it's been since the last release, but there were so many aspects of GSConnect I was unhappy with that I started nearly complete rewrite that's taken much longer than I hoped, which has also kept me from being as responsive to bug reports as I'd like.

In my current local branch quite a bit has changed with the Telephony plugin and messaging UI. 

![screenshot from 2018-02-26 19-03-53](https://user-images.githubusercontent.com/1265208/36708520-3796fe7e-1b28-11e8-96a5-0f76122b411b.png)

Some things that have been changed (but may change further):

* It's now possible to choose contact colors using the standard color picker (later possibly photos, depending on improvements folks integration), and the foreground color is automatically adjusted to be WCAG compliant. By default, contact colors will be truly random across the full the spectrum. I hope this will solve issues with readability, themes and so on.
* Additionally, it will be possible to delete "unknown" contacts that have been populated from incoming SMS/phone calls, and a mildy useful "open in Gnome Contacts" button if using Folks integration.
* The local address book isn't cached per-device anymore, since it will be the same anyways. There is in fact a [WIP contacts plugin](https://phabricator.kde.org/D9690) in the KDE Connect Android app, but it's nowhere near complete and so not yet implemented.
* Multi-recipient conversations are gone and there will instead be a "batch send" send feature (eg. for message such as "Seasons greetings, and the best to your family"). This is somewhat unfortunate, but not technically supported by SMS, and applications I've seen that do support it really just emulate it with nightmarish chains of if-then's and time-based guesswork.
* There will be timestamps and likely a rudimentary conversation log, although since full conversation logs are not available from the protocol, it won't be anything special.
* There will be much better control of how incoming messages are handled, especially with regard to the action taken when received (eg, auto-reply/open new window, show notification, custom command, etc). In fact, it will be possible to offload many plugins _purely_ to DBus, allowing plugin and event specific clients to be written, integrated into other programs, or handled completely outside of GSConnect.

And that's just messaging :) I put in as much time as I have available, but I'm honestly not sure how long it will be until v10 is released.

**Comment by ferdnyc** [comment]
Good stuff! Looking forward to it.

> There will be timestamps and likely a rudimentary conversation log, although since full conversation logs are not available from the protocol, it won't be anything special.

Yeah, history is definitely a rough point. For me it's actually most problematic that messages I _send_ don't appear in the Messages app on the device, which (otherwise) does a good job of serving as the primary SMS log. But, of course, that would have to be solved at the KDE Connect app end, not the GSConnect end… and I have my doubts as to whether it actually could be solved. 

(I'm trying to remember if Pushbullet somehow injected outgoing SMS content into the device Messages history. I don't _believe_ so, but it's been a while now. I know that [SMS Backup+](github.com/jberkel/sms-backup-plus) needs to temporarily become the default SMS app for the device, in order to restore backed-up messages into the local history. I suspect that would be true for any other app that wanted similar access.)

**Comment by andyholmes** [comment]
> For me it's actually most problematic that messages I send don't appear in the Messages app on the device... 

I've heard reports of this problem elsewhere, although I've never encountered it myself. I know that KDE Connect struggles with a number issues regarding the spread of API versions they support. Personally, my daily phone is still my old reliable Nexus 4 and so I'm using an AOSP Nougat build of some sort, which may make a difference. 

> ...that would have to be solved at the KDE Connect app end, not the GSConnect

Another possibility is the phone number discrepancy, which might be solvable in GSConnect. It's been awhile since I went through the KDE Connect desktop code, but the phone number the app sends when receiving telephony events is not always the same as Folks/GData reports (eg. the number is stripped of the international prefix such as `+001` or non-numeric characters as in `1-(555)`). GSConnect does it's best to respond to the same number it receives, but that's obviously not possible if it's the first to initiate contact. So it could just be a comparison problem.

Who knows, perhaps the WIP contacts plugin will be the solution. You're welcome to open an issue for that so I remember to come back to it when I have time, since some of these issues I just don't encounter.

**Comment by andyholmes** [comment]
I'm going to close this since the issue has been addressed in master. Colors for contact avatars are generated from the contact name, like Gnome Contacts does, and chat bubbles use theme defined colors:

![screenshot from 2018-09-25 21-37-39](https://user-images.githubusercontent.com/1265208/46057621-5afaff80-c10b-11e8-9671-5b24d50d3755.png)

The chat bubble and style can be overridden by themes, too.

**Comment by ferdnyc** [comment]
I'll be honest, since the launch of Android Messages' web interface I hardly ever use GSConnect's SMS features. But it sounds like a good solution!
